### ISSUE TITLE
Connection to port 8097 fails, failed to send WebSocket frame

### ISSUE THREAD

**Issue by michaelaflores** [facebook/react-devtools#841]
When I run `react-devtools`, it seems to have trouble connecting as it keeps changing from "Waiting for React to connect..." to "Connecting to React..." and back. The connection message typically flips back pretty quickly, and then throws this error to the console:

> WebSocket connection to 'ws://localhost:8097/' failed: Failed to send WebSocket frame.

It will go through this loop quite a few times (sometimes up to 60, sometimes just a few times) before successfully establishing a connection. If I use the Chrome extension, that works fine, but this standalone shell seems to have trouble connecting when I try to debug on Safari. Any help is definitely appreciated, let me know if I can provide more info.

System details:
macOS 10.12.5
react-devtools@2.5.0 
Safari 10.1.1

Index.html:
```
<!doctype html>
<html lang="en">
  <head>
    <script src="http://192.168.39.243:8097"></script>
```
Changing the script to `<script src="http://localhost:8097"></script>` has no impact on the issue.

**Comment by michaelaflores** [comment]
Update: opening the same page in Safari Technology Preview or Chrome allows the app shell to load as it should. I'm curious what about Safari specifically might cause this issue.

**Comment by bvaughn** [comment]
Off the top of my head I have no idea what this issue is. I don't think it's super high priority from our end if it impacts Safari only (and not even the tech preview). If you have time to dig into it though we'll be happy to review a PR. 😄

**Comment by michaelaflores** [comment]
I'd definitely be happy to dig in to it. Any links to how to get some debug info or get more info on the error would be appreciated. 😃

**Comment by msgirlperl** [comment]
I am having this issue as well, and I need to use Safari to debug a cordova / iPad issue.  I tried using both the script tag in my page and the import statement but both are generating the error. @michaelaflores, did you ever figure this out?  I'm happy to help as well if you have any other leads.

**Comment by SG5** [comment]
> I'd definitely be happy to dig in to it. Any links to how to get some debug info or get more info on the error would be appreciated. 😃

Try to run `react-devtools` on PC and open web site from phone

**Comment by natebowang** [comment]
For someone need a hot-fix for v3.6.1:
replace only one "localhost" with your PC's IP like "192.168.0.10" in this file: 
/usr/lib/node_modules/react-devtools/node_modules/react-devtools-core/build/backend.js

**Comment by bvaughn** [comment]
React DevTools has been rewritten and [recently launched a new version 4 UI](https://reactjs.org/blog/2019/08/15/new-react-devtools.html). The source code for this rewrite was done in a separate repository and now lives in the main React repo ([github.com/facebook/react](https://github.com/facebook/react/)).

Because version 4 was a total rewrite, and all issues in this repository are related to the old version 3 of the extension, I am closing all issues in this repository. If you can still reproduce this issue, or believe this feature request is still relevant, please open a new issue in the React repo: https://github.com/facebook/react/issues/new?labels=Component:%20Developer%20Tools
